smurfs/src/reducers/index.js

```diff
@@ -1,23 +1,20 @@
-/*
-  Be sure to import in all of the action types from `../actions`
-*/
+import { LOADING, GET_SMURFS, ERROR_MESSAGE } from '../actions/index';
 
-/*
- Your initial/default state for this project could *Although does not have to* look a lot like this
- {
-   smurfs: [],
-   fetchingSmurfs: false
-   addingSmurf: false
-   updatingSmurf: false
-   deletingSmurf: false
-   error: null
- }
-*/
+const defaultState = {
+  smurfs: [],
+  error: '',
+  loading: false,
+}
 
-/*
-  You'll only need one smurf reducer for this project.
-  Feel free to export it as a default and import as rootReducer. 
-  This will guard your namespacing issues.
-  There is no need for 'combineReducers' in this project.
-  Components can then read your store as, `state` and not `state.fooReducer`.
-*/
+export const rootReducer = (state = defaultState, action) => {
+  switch (action.type) {
+    case LOADING:
+      return Object.assign({}, state, { loading: true })
+    case GET_SMURFS:
+      return Object.assign({}, state, { smurfs: action.smurfs, loading: false, error: '' })
+    case ERROR_MESSAGE:
+      return Object.assign({}, state, { error: action.errorMessage, loading: false })
+    default:
+      return state
+  }
+}
```

Comment:
This reducer is extremely DRY, so that's fantastic. We did talk about using a spread-operator here instead of Object.assign(), but this does still work, so it's not a big deal.